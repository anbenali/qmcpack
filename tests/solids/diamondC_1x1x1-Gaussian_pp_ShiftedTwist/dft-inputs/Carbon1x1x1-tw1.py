#!/usr/bin/env python
import numpy
import h5py
from pyscf.pbc import gto, scf, dft
from pyscf.pbc import df 

cell = gto.Cell()
cell.a             = '''
         3.37316115       3.37316115       0.00000000
         0.00000000       3.37316115       3.37316115
         3.37316115       0.00000000       3.37316115'''
cell.atom = '''  
   C        0.00000000       0.00000000       0.00000000
   C        1.686580575      1.686580575      1.686580575 
            ''' 
cell.basis         = 'bfd-vdz'
cell.ecp           = 'bfd'
cell.unit          = 'B'
cell.drop_exponent = 0.15
cell.verbose       = 5
cell.charge        = 0
cell.spin          = 0
cell.build()

sp_twist=[0.333,0.333,0.333]
twist = numpy.asarray([0.333,0.333,0.333]) / 1.0
kmesh=[1,1,1]
kpts = cell.make_kpts((1,1,1), with_gamma_point=False,  wrap_around=True, scaled_center=twist)

print kpts
#axes = numpy.eye(3)*3.5668
#kaxes = 2*numpy.pi*numpy.linalg.inv(axes).T
#print (numpy.dot(kpts,axes)/numpy.pi/2)[13]
#exit()

mf = scf.KRHF(cell,kpts)
#mf.xc = 'pbe,pbe'

mf.exxdiv = 'ewald'

mf.max_cycle = 200




e_scf=mf.kernel()

ener = open('e_scf','w')
ener.write('%s\n' % (e_scf))
print('e_scf',e_scf)
ener.close()

title="S1-twist1"
from PyscfToQmcpack import savetoqmcpack
savetoqmcpack(cell,mf,title=title,kmesh=kmesh,kpts=kpts,sp_twist=kpts)

