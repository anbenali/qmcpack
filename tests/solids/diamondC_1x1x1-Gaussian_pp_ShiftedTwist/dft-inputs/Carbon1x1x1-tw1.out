Discarded 14 diffused primitive functions, 0 contracted functions
#INFO: **** input file is /home/abenali/Work/Development/QMCPACK-Benali/qmcpack/tests/solids/diamondC_1x1x1-Gaussian_pp_ShiftedTwist/dft-inputs/Carbon1x1x1-tw1.py ****
#!/usr/bin/env python
import numpy
import h5py
from pyscf.pbc import gto, scf, dft
from pyscf.pbc import df 

cell = gto.Cell()
cell.a             = '''
         3.37316115       3.37316115       0.00000000
         0.00000000       3.37316115       3.37316115
         3.37316115       0.00000000       3.37316115'''
cell.atom = '''  
   C        0.00000000       0.00000000       0.00000000
   C        1.686580575      1.686580575      1.686580575 
            ''' 
cell.basis         = 'bfd-vdz'
cell.ecp           = 'bfd'
cell.unit          = 'B'
cell.drop_exponent = 0.15
cell.verbose       = 5
cell.charge        = 0
cell.spin          = 0
cell.build()

sp_twist=[0.333,0.333,0.333]
twist = numpy.asarray([0.333,0.333,0.333]) / 1.0
kmesh=[1,1,1]
kpts = cell.make_kpts((1,1,1), with_gamma_point=False,  wrap_around=True, scaled_center=twist)

print kpts
#axes = numpy.eye(3)*3.5668
#kaxes = 2*numpy.pi*numpy.linalg.inv(axes).T
#print (numpy.dot(kpts,axes)/numpy.pi/2)[13]
#exit()

mf = scf.KRHF(cell,kpts)
#mf.xc = 'pbe,pbe'

mf.exxdiv = 'ewald'

mf.max_cycle = 200




e_scf=mf.kernel()

ener = open('e_scf','w')
ener.write('%s\n' % (e_scf))
print('e_scf',e_scf)
ener.close()

title="S1-twist1"
from PyscfToQmcpack import savetoqmcpack
savetoqmcpack(cell,mf,title=title,kmesh=kmesh,kpts=kpts,sp_twist=kpts)

#INFO: ******************** input file end ********************


System: ('Linux', 'abenali', '4.15.0-55-generic', '#60-Ubuntu SMP Tue Jul 2 18:22:20 UTC 2019', 'x86_64', 'x86_64')  Threads 8
Python 2.7.15+ (default, Nov 27 2018, 23:36:35) 
[GCC 7.3.0]
numpy 1.14.0  scipy 1.0.0
Date: Wed Aug  7 13:31:28 2019
PySCF version 1.7.1
PySCF path  /home/abenali/Work/src/pyscf/pyscf
GIT ORIG_HEAD 6ba52f91efc4c372028067d89a3410cbf6c4f744
GIT HEAD      ref: refs/heads/dev
GIT dev branch  262e23471167864e983dfa0b7c13471750037576

[CONFIG] conf_file None
[INPUT] verbose = 5
[INPUT] max_memory = 4000 
[INPUT] num. atoms = 2
[INPUT] num. electrons = 8
[INPUT] charge = 0
[INPUT] spin (= nelec alpha-beta = 2S) = 0
[INPUT] symmetry False subgroup None
[INPUT] Mole.unit = B
[INPUT]  1 C      0.000000000000   0.000000000000   0.000000000000 AA    0.000000000000   0.000000000000   0.000000000000 Bohr
[INPUT]  2 C      0.892500004670   0.892500004670   0.892500004670 AA    1.686580575000   1.686580575000   1.686580575000 Bohr
[INPUT] ---------------- BASIS SET ---------------- 
[INPUT] l, kappa, [nprim/nctr], expnt,             c_1 c_2 ...
[INPUT] C
[INPUT] 0    0    [7    /1   ]  13.073594         -0.000705
                                6.541187          0.015176
                                3.272791          -0.121499
                                1.637494          -0.033512
                                0.819297          0.180113
                                0.409924          0.380369
                                0.2051            0.397529
[INPUT] 1    0    [6    /1   ]  7.480076          0.021931
                                3.741035          0.054425
                                1.871016          0.112024
                                0.935757          0.210979
                                0.468003          0.289868
                                0.234064          0.302667
[INPUT] 2    0    [1    /1   ]  0.561161             1

Ewald components = 8.63915484117797e-26, -47.334679725158, 34.559012492297
nuclear repulsion = -12.7756672328609
number of shells = 6
number of NR pGTOs = 60
number of NR cGTOs = 18
basis = bfd-vdz
ecp = bfd
CPU time:         1.81
lattice vectors  a1 [3.373161150, 3.373161150, 0.000000000]
                 a2 [0.000000000, 3.373161150, 3.373161150]
                 a3 [3.373161150, 0.000000000, 3.373161150]
dimension = 3
low_dim_ft_type = None
Cell volume = 76.7611
exp_to_discard = 0.15
rcut = 14.27373091585211 (nimgs = [4 4 4])
lattice sum = 455 cells
precision = 1e-08
pseudo = None
ke_cutoff = 366.8014593069121
    = [34 34 34] mesh (39304 PWs)
ew_eta = 2.61119
ew_cut = 2.673899610122948 (nimgs = [1 1 1])
[[0.31013945 0.31013945 0.31013945]]


******** <class 'pyscf.pbc.scf.khf.KRHF'> ********
method = KRHF-KSCF-RHF-SCF-RHF
initial guess = minao
damping factor = 0
level shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
DIIS start cycle = 1
DIIS space = 8
SCF tol = 1e-07
SCF gradient tol = None
max. SCF cycles = 200
direct_scf = False
chkfile to save SCF result = /home/abenali/Work/Development/QMCPACK-Benali/qmcpack/tests/solids/diamondC_1x1x1-Gaussian_pp_ShiftedTwist/dft-inputs/tmpRdTLAm
max_memory 4000 MB (current use 66 MB)


******** PBC SCF flags ********
N kpts = 1
kpts = [[0.31013945 0.31013945 0.31013945]]
Exchange divergence treatment (exxdiv) = ewald
Ewald components = 2.3382937731203e-69, -1.47620749513098, 1.13640294140598
    madelung (= occupied orbital energy shift) = 0.6796091074500152
    Total energy shift due to Ewald probe charge = -1/2 * Nelec*madelung = -2.7184364298
DF object = <pyscf.pbc.df.fft.FFTDF object at 0x149d2cf1fcd0>


******** <class 'pyscf.pbc.df.fft.FFTDF'> ********
mesh = [34 34 34] (39304 PWs)
len(kpts) = 1
Set gradient conv threshold to 0.000316228
Big error detected in the electron number of initial guess density matrix (Ne/cell = 8.26837)!
  This can cause huge error in Fock matrix and lead to instability in SCF for low-dimensional systems.
  DM is normalized wrt the number of electrons 8.0
cond(S) = [17.20410133]
Ewald components = 2.3382937731203e-69, -1.47620749513098, 1.13640294140598
    CPU time for vj and vk     44.14 sec, wall time      5.61 sec
Ewald components = 8.63915484117797e-26, -47.334679725158, 34.559012492297
E1 = (3.9229376892666536-4.272139938331215e-16j)  E_coul = (-3.121579915982525+3.0369072345940743e-16j)
init E= -11.9743094595768
    CPU time for initialize scf    100.72 sec, wall time     12.82 sec
HOMO = 0.516925496432  LUMO = 1.03157062446
     k-point                  mo_energy
   0 ( 0.333  0.333  0.333)   [-1.151177   0.0064893  0.5169255  0.5169255] [1.03157062 1.05861694 1.05861694 1.41699738 1.85114796 1.85114796
 2.3887833  2.43569382 2.43569382 2.68591355 3.02105561 3.02105561
 3.51249538 3.51249538]
Ewald components = 2.3382937731203e-69, -1.47620749513098, 1.13640294140598
    CPU time for vj and vk     26.05 sec, wall time      3.30 sec
Ewald components = 8.63915484117797e-26, -47.334679725158, 34.559012492297
E1 = (4.147931377685451-4.3259666682171627e-16j)  E_coul = (-2.5473395161560086+2.054563116860031e-16j)
cycle= 1 E= -11.1750753713315  delta_E= 0.799  |g|= 0.127  |ddm|= 2.79
    CPU time for cycle= 1     26.57 sec, wall time      3.37 sec
HOMO = 0.113095673035  LUMO = 1.16824094883
     k-point                  mo_energy
   0 ( 0.333  0.333  0.333)   [-0.53132624 -0.17016556  0.11309567  0.11309567] [1.16824095 1.16824095 1.19333693 1.56077187 1.87129683 1.87129683
 2.41193542 2.43072847 2.43072847 2.69971252 3.03515239 3.03515239
 3.50913714 3.50913714]
Ewald components = 2.3382937731203e-69, -1.47620749513098, 1.13640294140598
    CPU time for vj and vk     22.33 sec, wall time      2.84 sec
Ewald components = 8.63915484117797e-26, -47.334679725158, 34.559012492297
E1 = (4.151198952252218-4.369334755116583e-16j)  E_coul = (-2.555457620077709+2.865546341879188e-16j)
cycle= 2 E= -11.1799259006864  delta_E= -0.00485  |g|= 0.0341  |ddm|= 0.121
    CPU time for cycle= 2     22.95 sec, wall time      2.92 sec
HOMO = 0.112411009753  LUMO = 1.1692779678
     k-point                  mo_energy
   0 ( 0.333  0.333  0.333)   [-0.53168742 -0.17445852  0.11241101  0.11241101] [1.16927797 1.16927797 1.19757622 1.55592292 1.86860046 1.86860046
 2.41010077 2.42835924 2.42835924 2.69545616 3.03497255 3.03497255
 3.51000228 3.51000228]
Ewald components = 2.3382937731203e-69, -1.47620749513098, 1.13640294140598
    CPU time for vj and vk     26.32 sec, wall time      3.34 sec
Ewald components = 8.63915484117797e-26, -47.334679725158, 34.559012492297
E1 = (4.152993046946407-4.47124975933022e-16j)  E_coul = (-2.5577302921485194+8.673617379884035e-17j)
cycle= 3 E= -11.1804044780631  delta_E= -0.000479  |g|= 0.00696  |ddm|= 0.0498
    CPU time for cycle= 3     27.03 sec, wall time      3.44 sec
HOMO = 0.112273893023  LUMO = 1.16903303434
     k-point                  mo_energy
   0 ( 0.333  0.333  0.333)   [-0.53164331 -0.17433971  0.11227389  0.11227389] [1.16903303 1.16903303 1.19733783 1.55567272 1.86854169 1.86854169
 2.41011228 2.42830229 2.42830229 2.69536621 3.03472783 3.03472783
 3.5098083  3.5098083 ]
Ewald components = 2.3382937731203e-69, -1.47620749513098, 1.13640294140598
    CPU time for vj and vk     25.68 sec, wall time      3.26 sec
Ewald components = 8.63915484117797e-26, -47.334679725158, 34.559012492297
E1 = (4.153287659319109-4.378008372496467e-16j)  E_coul = (-2.5580534871903713+2.9219248548484344e-16j)
cycle= 4 E= -11.1804330607322  delta_E= -2.86e-05  |g|= 0.00137  |ddm|= 0.0147
    CPU time for cycle= 4     25.92 sec, wall time      3.29 sec
HOMO = 0.11229932612  LUMO = 1.16902085061
     k-point                  mo_energy
   0 ( 0.333  0.333  0.333)   [-0.53162925 -0.17437248  0.11229933  0.11229933] [1.16902085 1.16902085 1.19739995 1.55565185 1.86851852 1.86851852
 2.41009399 2.42831509 2.42831509 2.69537463 3.03472353 3.03472353
 3.50983218 3.50983218]
Ewald components = 2.3382937731203e-69, -1.47620749513098, 1.13640294140598
    CPU time for vj and vk     27.19 sec, wall time      3.44 sec
Ewald components = 8.63915484117797e-26, -47.334679725158, 34.559012492297
E1 = (4.153251348639258-4.371503159461554e-16j)  E_coul = (-2.558018397897859+2.8210940528072825e-16j)
cycle= 5 E= -11.1804342821195  delta_E= -1.22e-06  |g|= 0.000325  |ddm|= 0.00271
    CPU time for cycle= 5     27.89 sec, wall time      3.53 sec
HOMO = 0.112301387617  LUMO = 1.16903077727
     k-point                  mo_energy
   0 ( 0.333  0.333  0.333)   [-0.53161962 -0.17436609  0.11230139  0.11230139] [1.16903078 1.16903078 1.19740065 1.55565796 1.86852067 1.86852067
 2.41009813 2.42832403 2.42832403 2.69537567 3.03472406 3.03472406
 3.50984441 3.50984441]
Ewald components = 2.3382937731203e-69, -1.47620749513098, 1.13640294140598
    CPU time for vj and vk     27.07 sec, wall time      3.42 sec
Ewald components = 8.63915484117797e-26, -47.334679725158, 34.559012492297
E1 = (4.153230235330598-4.38451358553138e-16j)  E_coul = (-2.557997378001595+2.809167828909942e-16j)
cycle= 6 E= -11.1804343755319  delta_E= -9.34e-08  |g|= 6.21e-05  |ddm|= 0.000818
    CPU time for cycle= 6     27.66 sec, wall time      3.50 sec
HOMO = 0.112303269268  LUMO = 1.1690332527
     k-point                  mo_energy
   0 ( 0.333  0.333  0.333)   [-0.53162061 -0.17436819  0.11230327  0.11230327] [1.16903325 1.16903325 1.19740796 1.55566154 1.86852115 1.86852115
 2.41009931 2.42832674 2.42832674 2.69537791 3.03472633 3.03472633
 3.50984733 3.50984733]
Ewald components = 2.3382937731203e-69, -1.47620749513098, 1.13640294140598
    CPU time for vj and vk     21.74 sec, wall time      2.75 sec
Ewald components = 8.63915484117797e-26, -47.334679725158, 34.559012492297
E1 = (4.153224244492605-4.499439015814843e-16j)  E_coul = (-2.557991389517255+7.827939685345342e-17j)
Extra cycle  E= -11.1804343778856  delta_E= -2.35e-09  |g|= 3.13e-05  |ddm|= 7.6e-05
    CPU time for scf_cycle    280.93 sec, wall time     35.69 sec
    CPU time for SCF    281.67 sec, wall time     35.79 sec
converged SCF energy = -11.1804343778856
('e_scf', -11.180434377885597)
Using Python 2.x
Wavefunction successfully saved to QMCPACK HDF5 Format
Use: "convert4qmc -pyscf  S1-twist1.h5" to generate QMCPACK input files
